src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate(operation, a) {
+      operation.call(this, a);
+
+      return this;
+    },
+
+    add(a) {
+      this.result += a;
+    },
+
+    subtract(a) {
+      this.result -= a;
+    },
+
+    multiply(a) {
+      this.result *= a;
+    },
+
+    divide(a) {
+      this.result /= a;
```

Comment:
No handling for division by zero. This will cause an Infinity result or error. Add a condition to check if the divisor 'a' is zero before performing the operation.